gvsbuild/utils/builder.py

```diff
@@ -105,10 +112,25 @@ def __check_vs(self, opts):
         self.add_env('LIBPATH', os.path.join(self.gtk_dir, 'lib'))
         self.add_env('PATH', os.path.join(self.gtk_dir, 'bin'))
 
-        output = subprocess.check_output('cmd.exe /c ""%s" && set"' % (vcvars_bat,), shell=True)
+        temp_env_out_path = tempfile.mktemp()
+        try:
+            # Use output redirect will help to ignore the header part vcvars family print
+            # Which make later parser more reliable
+            cmd_details='cmd.exe /c ""%s" && set>%s"' % (vcvars_bat, temp_env_out_path)
+            retcode = subprocess.call(cmd_details, shell=True)
+            if retcode==0:
+                with open(temp_env_out_path) as temp_env_out_file:
+                    output=temp_env_out_file.read()
+            else:
+                raise Exception("Some error happened when try to run vcvars family, cmd: %s, cwd: %s" % (cmd_details, os.getcwd()))

```

Comment:
the self provide exception can provide more details for easy debug, the default CalledProcessError like 

 subprocess.CalledProcessError: Command 'exit 1' returned non-zero exit status 1

user does not know where the command be called, the cwd info is very important for debug.